networkx/algorithms/community/modularity_max.py

```diff
@@ -120,10 +108,11 @@ def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1)
 
     # Initialize single-node communities
     communities = {n: frozenset([n]) for n in G}
+    yield sorted(communities.values(), key=len, reverse=True)
```

Comment:
Does this sort add significantly to the time? I would think it could be a slow step.
The user can sort the yielded communities by length if needed -- and not if they don't want to spend the cpu time doing that.  

Maybe:  `yield list(communities.values()`